schema/vrs.yaml

```diff
@@ -396,27 +387,68 @@ definitions:
     required: [ "type", "gene_id" ]
 
   # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-  # Range
+  # Numerics, Comparators, and Ranges
 
   Range:
     description: >-
-      An inclusive range of integers. A range may be unbounded by omitting either
-      min or max values. Fully unbounded ranges are not permitted (one of min or
-      max MUST be specified).
+      A set of classes that provide the possible representations of integer number
+      ranges, from a single integer (BoundedInteger) to bounded ranges (ClosedRange)
+      or half-bounded ranges (OpenRange).
+    oneOf:
```

Comment:
I'll only whine a little here since you are doing such fantastic work. But I'm not a fan of the names.
There's an unnatural tension between calling a single precise value a "kind of" Range just as much as there is calling a min and max range a "kind of" Integer, point, count, etc.. 

I  keep going back to the term "Quantity" which is the amount of something. And while I agree it seems perverse to think of a position on a coordinate system as a Quantity, it does feel okay calling a single fixed value as well as a range of values (open or closed) as a quantity, since quantities conceptually can be precise or ambiguously referenced.

We may simply want to NOT try so hard to reuse this general-purpose data type for both coordinate positions and for quantities (like counts and measures).  While I agree it would be nice to not replicate two very similar patterns it seems like we keep going in circles with this.  I could live with using the term "Quantity" for the position on a coordinate system since it can be construed as the amount of units that the position represents.

Then SimpleQuantity, UnboundedQuantity and Range serve the purposes of both Endpoints on a sequence location as well as  Counts in a copy number abundance concept. And it would set the stage to use these in other measures defined within systems - either coordinate-wise or amount-wise.

But since the term Quantity is a non-started (I assume) let's try something else...

How about this

IntegerValue (abstract class name)
* FixedInteger (or SimpleInteger)
* OpenIntegerRange
* IntegerRange (or ClosedIntegerRange)

I still think the first two (FixedInteger & OpenIntegerRange) could share a common parent since they both have a base-value attribute and the OpenIntegerRange has only the addition of a comparator attribute. But I really don't see the justification, value or logic in having the IntegerRange inherit from the same parent as those two unless one of the two (min or max) is going to be the shared "base-value" that is inherently represented by the ancestor concept and the other is an outer limit. Keeping with the min/max model for Closed Range does mean that both min and max are conceptually identical as the Fixed or Simple Integer.

Again, apologies for the monkey wrench, but I feel like we are so close and I'd like to see if there's any changes in the understanding of our collective that we may all see some of these arguments in a better light now.

Or can counter args be made to help bring me closer to the fold.

Reply:
'OpenIntegerRange` would imply the interval `(a, b]`, `[a, b)` or `(a, b)` i.e. left-open, right-open or fully-open to someone not reading the docs and it's not immediately obvious from the name either.

Reply:
@larrybabb this is no problem. I will revise to align more closely to our previous work, starting from the names you suggest here. I do agree with @julesjacobsen that the use of `Open` / `Closed` conflicts with the usage of those descriptions [in other contexts](https://en.wikipedia.org/wiki/Interval_(mathematics)), so perhaps we could have:

- `SimpleInteger`
- `HalfBoundedIntegerRange` (instead of `OpenIntegerRange`)
- `BoundedIntegerRange` (instead of `ClosedIntegerRange`)

Since we don't actually have inheritance _per se_ using JSON Schema draft 7, I think we should de-emphasize concerns about attribute inheritance patterns. Instead, I think it would be reasonable for us to create a conceptual grouping to contain all three of these classes (similar to our previous conceptual groupings such as `MolecularVariation`). This grouping could be something along the lines of `IntegerOrRange`, and would serve as a a single reference we could use whenever any of these three classes are acceptable, resulting in:

- `IntegerOrRange`
  - `SimpleInteger`
  - `HalfBoundedIntegerRange`
  - `BoundedIntegerRange`

How does this sound?